src/ape/api/providers.py

```diff
@@ -95,30 +76,33 @@ def total_transfer_value(self) -> int:
         Useful for determining if an account balance can afford
         to submit the transaction.
         """
-        return self.value + self.max_fee
+        if self.max_fee:
+            return self.value + self.max_fee
+
+        raise TransactionError(message="Max fee must not be null.")
 
     @abstractmethod
-    def encode(self) -> bytes:
+    def serialize_transaction(self) -> bytes:
         """
         Take this object and produce a hash to sign to submit a transaction
         """
 
-    def as_dict(self) -> dict:
+    def as_dict(self, exclude: Optional[Set[str]] = None) -> dict:
         """
         Create a ``dict`` representation of the transaction.
 
         Returns:
             dict
         """
-        return as_dict(self)
```

Comment:
Here is where we should also remove `from`.
`from` seems to cause issues down the line, I have to remove it here to get anything to work.

```python
        data = self.dict(by_alias=True, exclude_none=True, exclude=exclude)

        # Don't publish sender
        if "from" in data:
            del data["from"]

        return data

```

Reply:
I am not 100% sure on this but somewhere, `from` needs to either be removed or set to its proper value, specifically for calls (not really transactions, although needs more testing)